test/src/test/java/hudson/model/ItemGroupMixInTest.java

```diff
@@ -47,4 +55,76 @@
         assertEquals("valid", items.iterator().next().getName());
     }
 
+  /**
+   * This test unit makes sure that if part of the config.xml file is
+   * deleted it will still load everything else inside the folder.
+   * The test unit expects an IOException is thrown, and the one failed
+   * job fails to load.
+   */
+  @Issue("JENKINS-22811")
+  @Test
+  public void xmlFileFailsToLoad() throws Exception {
+    MockFolder folder = r.createFolder("folder");
+    assertNotNull(folder);
+
+    AbstractProject project = folder.createProject(FreeStyleProject.class, "job1");
+    AbstractProject project2 = folder.createProject(FreeStyleProject.class, "job2");
+    AbstractProject project3 = folder.createProject(FreeStyleProject.class, "job3");
+
+    File configFile = project.getConfigFile().getFile();
+
+    List<String> lines = FileUtils.readLines(configFile).subList(0, 5);
+    configFile.delete();
+
+    // Remove half of the config.xml file to make "invalid" or fail to load
+    FileUtils.writeByteArrayToFile(configFile, lines.toString().getBytes());
+    for (int i = lines.size() / 2; i < lines.size(); i++) {
+      FileUtils.writeStringToFile(configFile, lines.get(i), true);
+    }
+
+    // Reload Jenkins.
+    r.jenkins.reload();
+
+    // Folder
+    assertNotNull("Folder failed to load.", folder);
+    assertNull("Job should have failed to load.", r.jenkins.getItemByFullName("folder/job1"));
+    assertNotNull("Other job in folder should have loaded.", r.jenkins.getItemByFullName("folder/job2"));
+    assertNotNull("Other job in folder should have loaded.", r.jenkins.getItemByFullName("folder/job3"));
+  }
+
+  /**
+   * This test unit makes sure that any exception that is thrown is caught, however the folder
+   * will continue to load not impacting any other jobs in the folder.
+   */
+  @LocalData
+  @Issue("JENKINS-22811")
+  @Test
+  public void xmlFileReadExceptionOnLoad() throws Exception {
+    MockFolder d = r.jenkins.getItemByFullName("d", MockFolder.class);
+    assertNotNull(d);
+    Collection<TopLevelItem> items = d.getItems();
+    assertEquals(1, items.size());
+    assertEquals("valid", items.iterator().next().getName());
+  }
+
+  @Extension
+  public static class MockBuilderThrowsError extends BuildWrapper {
+    @Override
+    public Collection<? extends Action> getProjectActions(AbstractProject project){
+      throw new NullPointerException();
```

Comment:
You did not read [my comment](https://issues.jenkins-ci.org/browse/JENKINS-22811?focusedCommentId=217345&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-217345). Even if `getProjectActions` throws an (unchecked) exception, the job should still be loaded, with the exception logged. And of course `xmlFileReadExceptionOnLoad` should then be asserting that _both_ jobs are present.